### ISSUE TITLE
Incorrect no-undef error 

### ISSUE THREAD

**Issue by bjrmatos** [eslint/eslint#4464]
i have some code:

```js
let { output, ...restData } = childData;

debugStrategy('conversion ended successfully..');

cb(null, {
   ...restData,
   stream: fs.createReadStream(output)
});
```

and I'm getting an error -> "restData" is not defined

<img width="531" alt="captura de pantalla 2015-11-17 a las 2 20 34 p m" src="https://cloud.githubusercontent.com/assets/4262050/11222076/68170760-8d36-11e5-8e18-59ab7c5ca7b9.png">

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by bjrmatos** [comment]
eslint version: 1.9.0

**Comment by bjrmatos** [comment]
my .eslintrc file

```js
{
  "extends": "eslint-config-airbnb/base", // https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/.eslintrc

  // Allow the following global variables
  "env": {
    "browser": true, // browser global variables
    "node": true, // Node.js global variables and Node.js scoping.
    "mocha": true // adds all of the Mocha testing global variables.
  },

  "ecmaFeatures": {
    "restParams": true,
    "experimentalObjectRestSpread": true
  },

  "rules": {
    /**
     * ES6
     */
    "prefer-const": 0,

    /**
     * Variables
     */
    "no-shadow": [2, {
      "builtinGlobals": true
    }],
    "no-unused-vars": [2, {
      "vars": "all",
      "args": "after-used"
    }],
    "no-use-before-define": [2, "nofunc"],

    /**
     * Possible errors
     */
    "comma-dangle": [2, "never"],
    "no-inner-declarations": [2, "both"],

    /**
     * Best practices
     */
    "curly": 2,
    "dot-notation": [2, {
      "allowKeywords": true,
      "allowPattern": "^[a-z]+(_[a-z]+)+$"
    }],
    "eqeqeq": [2, "allow-null"],
    "no-eq-null": 0,
    "no-redeclare": [2, {
      "builtinGlobals": true
    }],
    "wrap-iife": [2, "inside"],

    /**
     * Style
     */
    "indent": [2, 2, {
      "VariableDeclarator": {
        "var": 2,
        "let": 2,
        "const": 3
      }
    }],
    "func-names": 0,
    "no-multiple-empty-lines": [2, {
      "max": 1
    }],
    "no-extra-parens": [2, "functions"],
    "one-var": 0
  }
}

```

**Comment by nzakas** [comment]
It's because you're using the experimental object spread property. escope doesn't recognize experimental properties, so not sure if this is fixable.

**Comment by bjrmatos** [comment]
thnks, i will be waiting to know if this problem can be solved

**Comment by mysticatea** [comment]
I think this is duplicate of #3271.
I had fixed this in `escope`, but `escope` has not been released yet.

**Comment by nzakas** [comment]
Ah yes, closing in favor of #3271
